pkg/plugin/plugin.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package validator provides functions to validate jira issue against
+// validation criteria.
+
+// Package plugin provides the implementation of the JVS plugin interface.
+package plugin
+
+import (
+	"context"
+	"fmt"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"cloud.google.com/go/secretmanager/apiv1/secretmanagerpb"
+	"github.com/abcxyz/jvs-plugin-jira/pkg/validator"
+	jvspb "github.com/abcxyz/jvs/apis/v0"
+)
+
+const (
+	// jiraCategory is the justification category this plugin will be validating.
+	jiraCategory = "jira"
+)
+
+// issueMatcher is the mockable interface for the convenience of testing.
+type issueMatcher interface {
+	MatchIssue(context.Context, string) (*validator.MatchResult, error)
+}
+
+// JiraPlugin is the implementation of jvspb.Validator interface.
+type JiraPlugin struct {
+	validator issueMatcher
+}
+
+// NewJiraPlugin creates a new JiraPlugin.
+func NewJiraPlugin(ctx context.Context, cfg *PluginConfig) (*JiraPlugin, error) {
+	apiToken, err := secretVersion(ctx, cfg.APITokenSecretID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch API token: %w", err)
+	}
+
+	v, err := validator.NewValidator(cfg.JIRAEndpoint, cfg.Jql, cfg.JIRAAccount, apiToken)
+	if err != nil {
+		return nil, fmt.Errorf("failed to instantiate validator: %w", err)
+	}
+	return &JiraPlugin{
+		validator: v,
+	}, nil
+}
+
+// Validate returns the validation result.
+func (j *JiraPlugin) Validate(ctx context.Context, req *jvspb.ValidateJustificationRequest) (*jvspb.ValidateJustificationResponse, error) {
+	if got, want := req.Justification.Category, jiraCategory; got != want {
+		return nil,
+			fmt.Errorf(
+				"failed to perform validation, expected category %q to be %q",
```

Comment:
Nit: I'm surprised the linter isn't complaining, but this should be:

```suggestion
			fmt.Errorf("failed to perform validation, expected category %q to be %q",
```